tools/shared/src/main/scala/org/scalajs/core/tools/linker/Linker.scala

```diff
@@ -75,83 +70,12 @@ final class Linker(frontend: LinkerFrontend, backend: LinkerBackend)
-       *  On the JavaScript platform, this does not have any effect.
-       */
-      val closureCompilerIfAvailable: Boolean,
-      /** Additional configuration for the linker frontend. */
-      val frontendConfig: LinkerFrontend.Config,
-      /** Additional configuration for the linker backend. */
-      val backendConfig: LinkerBackend.Config
-  ) {
-    def withOptimizer(optimizer: Boolean): Config =
-      copy(optimizer = optimizer)
-
-    def withParallel(parallel: Boolean): Config =
-      copy(parallel = parallel)
-
-    def withClosureCompilerIfAvailable(closureCompilerIfAvailable: Boolean): Config =
-      copy(closureCompilerIfAvailable = closureCompilerIfAvailable)
-
-    def withFrontendConfig(frontendConfig: LinkerFrontend.Config): Config =
-      copy(frontendConfig = frontendConfig)
-
-    def withFrontendConfig(f: LinkerFrontend.Config => LinkerFrontend.Config): Config =
-      copy(frontendConfig = f(frontendConfig))
-
-    def withBackendConfig(backendConfig: LinkerBackend.Config): Config =
-      copy(backendConfig = backendConfig)
-
-    def withBackendConfig(f: LinkerBackend.Config => LinkerBackend.Config): Config =
-      copy(backendConfig = f(backendConfig))
-
-    private def copy(
-        optimizer: Boolean = optimizer,
-        parallel: Boolean = parallel,
-        closureCompilerIfAvailable: Boolean = closureCompilerIfAvailable,
-        frontendConfig: LinkerFrontend.Config = frontendConfig,
-        backendConfig: LinkerBackend.Config = backendConfig): Config = {
-      new Config(
-          optimizer = optimizer,
-          parallel = parallel,
-          closureCompilerIfAvailable = closureCompilerIfAvailable,
-          frontendConfig = frontendConfig,
-          backendConfig = backendConfig)
-    }
+object Linker {
+  def apply(frontendConfig: LinkerFrontend.Config,
+      backendConfig: LinkerBackend.Config): Linker = {
+    apply(LinkerFrontend(frontendConfig), LinkerBackend(backendConfig))
   }
 
-  object Config {
-    import LinkerPlatformExtensions._
-
-    implicit def toPlatformExtensions(config: Config): ConfigExt =
-      new ConfigExt(config)
-
-    /** Default configuration.
-     *
-     *  - `optimizer`: true
-     *  - `parallel`: true
-     *  - `closureCompilerIfAvailable`: false
-     *  - `frontendConfig`: default frontend configuration as returned by
-     *    [[org.scalajs.core.tools.linker.frontend.LinkerFrontend.Config.apply]]
-     *  - `backendConfig`: default backend configuration as returned by
-     *    [[org.scalajs.core.tools.linker.backend.LinkerBackend.Config.apply]]
-     */
-    def apply(): Config = {
-      new Config(
-          optimizer = true,
-          parallel = true,
-          closureCompilerIfAvailable = false,
-          frontendConfig = LinkerFrontend.Config(),
-          backendConfig = LinkerBackend.Config())
-    }
-  }
+  def apply(frontend: LinkerFrontend, backend: LinkerBackend): Linker =
+    new Linker(frontend, backend)
```

Comment:
Yes. The `class Linker` in this file is actually what will eventually become the class `standard.StandardLinker`, and this `apply` method will be moved to the object `standard.StandardLinker`. And the current `linker.GenLinker` will become `linker.Linker`.